class.rb

```diff
@@ -0,0 +1,7 @@
+class Class
+  def new(...)
+    obj = Primitive.rb_class_alloc2
+    Primitive.send_delegate!(obj, :initialize, ...)
```

Comment:
Does this mean there will be a single inline cache, so essentially a 1-entry global cache for calls to `initialize`?
Also I'm not sure but it might be updated on every call with a different receiver class, which would be expensive.
I guess currently the call to `initialize` from `new` is completely uncached (`rb_funcall`), so maybe it's about the same in this PR for that aspect?

Reply:
I realize on YJIT this inline cache concern might matter less because it creates its own inline cache by generating for the receiver it sees. Although that inline cache would only be per caller-of-new if YJIT always inlines `new`, otherwise it would still be global.

Reply:
> I guess currently the call to `initialize` from `new` is completely uncached (`rb_funcall`), so maybe it's about the same in this PR for that aspect?

The C calls use the global call cache.  I did [some refactoring on the global call cache](https://github.com/ruby/ruby/pull/10623), so we can use it for the initialize case.  This should make it hit cache as well as the C version did. See bb5fdce4d88d16472f7071e9b8156a12ff9eeab0

Reply:
Ah right, there is still the global cache, that's at least (in the hit case) one (simplified) hashmap lookup vs multiple.
FWIW, I looked, `Class#new` uses `rb_obj_call_init_kw()` which uses `rb_funcallv_kw()` and that doesn't have any inline cache.